### DISCUSSION TITLE
--hls-playlist-reload-attempts

### DISCUSSION THREAD

**Discussion by campones** [streamlink/streamlink#5293] [Q&A] [answered]
### Description

Hello

I wanted a stream to end up more quickly when the token expires so it could also start over faster too.

So I have been playing around with those 2 settings as follow:
--hls-playlist-reload-attempts 1
--stream-timeout 2.0

Initially in the logs,  I could count like 28 times the expression "failed to reload playlist" and now I have around 5 to 7.  It's much better but still I m not sure why I get so many retries after setting up a single attempt. Can I setup this to 0 as well?.

Thanks

**Comment by bastimeyer** [comment, marked as answer]
> --hls-playlist-reload-attempts

This is only relevant for HTTP error responses.

See issue #2198

**Comment by campones** [comment]
isn't 403 an error response? 

I m reading about this "hls-timeout" option but I couldn't see it in the command line section. 

ok so yea, you changed that option into stream-timeout but I m using this option already and I see at least 7 retries so what value should I use with this parameters to give up instantly as soon as the stream fails?

**Reply by bastimeyer** [reply]
> hls-timeout

This was removed
https://streamlink.github.io/deprecations.html#stream-type-related-cli-arguments

> 403

The number of HLS playlist reload attempts are only relevant for one playlist refresh. If there are any HTTP response errors, that one refresh will attempt as many times as defined by the parameter. If all attempts failed, then the worker thread will still continue. Just as it will continue when no new segments got queued. This queuing issue is what the linked issue thread is about.

**Comment by campones** [comment]
So i should use  either  --stream-segment-timeout and / or --stream-segment-attempts  I assume. 
I will give it a try thanks

**Reply by bastimeyer** [reply]
No, and stop asking. I have linked the relevant issue. You can't limit the reload attempts, you can only reload the `stream-timeout`, but this affects other things as well.

**Comment by campones** [comment]
so what's the reason to provide those cli option if they have no effects whatsoever? I have to say it's a bit confusing. 

I guess the goal of those options is to increase the default value so in case of failing streams, the process won't be ending. but lowering it won't make it stop faster then. 
then I kinda got my answer the stream-timeout that I m using is the only way to reduce this
